src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -73,27 +88,22 @@ internal class CompletionHandler : IRequestHandler<LSP.CompletionParams, LSP.Com
                 return null;
             }
 
-            var position = await document.GetPositionFromLinePositionAsync(ProtocolConversions.PositionToLinePosition(request.Position), cancellationToken).ConfigureAwait(false);
             var completionOptions = await GetCompletionOptionsAsync(document, cancellationToken).ConfigureAwait(false);
             var completionService = document.Project.LanguageServices.GetRequiredService<CompletionService>();
+            var documentText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
 
-            // TO-DO: More LSP.CompletionTriggerKind mappings are required to properly map to Roslyn CompletionTriggerKinds.
-            // https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1178726
-            var completionTrigger = await ProtocolConversions.LSPToRoslynCompletionTriggerAsync(request.Context, document, position, cancellationToken).ConfigureAwait(false);
-
-            var list = await completionService.GetCompletionsAsync(document, position, completionTrigger, options: completionOptions, cancellationToken: cancellationToken).ConfigureAwait(false);
-            if (list == null || list.Items.IsEmpty || cancellationToken.IsCancellationRequested)
+            var completionListResult = await GetCompletionListAndFilterDownAsync(request, documentText, document, completionOptions, completionService, cancellationToken).ConfigureAwait(false);
+            if (completionListResult == null)
             {
                 return null;
             }
 
+            var (filteredList, isInComplete, resultId) = completionListResult.Value;
```

Comment:
consider still calling this 'list'.  unless you think making it clear that it was filtered is very important for the code that follows.